tools/cosmovisor/go.mod

```diff
@@ -32,7 +32,7 @@ require (
 	github.com/btcsuite/btcd/btcec/v2 v2.3.2 // indirect
 	github.com/cespare/xxhash v1.1.0 // indirect
 	github.com/cespare/xxhash/v2 v2.1.2 // indirect
-	github.com/confio/ics23/go v0.7.0 // indirect
+	github.com/confio/ics23/go v0.9.0 // indirect
```

Comment:
I wonder if pushing the result of `go work sync` and running `./scripts/go-mod-tidy-all.sh` is correct.
I think we should not be pushing the result of `go work` as we use it only for development.
Here, cosmovisor depends on `github.com/cosmos/cosmos-sdk v0.46.0-beta2.0.20221027164535-be6b93d95c30`, so I don't see how `github.com/confio/ics23/go v0.9.0` can be an indirect dependency.

Can we maybe make this PR about `make proto-all`  /  `make proto-format`?

Reply:
sure, that is interesting!

what is off about:

```
github.com/confio/ics23/go v0.9.0
```

Reply:
I don't understand how it can be an indirect dependency. Given that noting in the `go.mod` depends on `github.com/confio/ics23/go v0.9.0`. As we purposely do not push the result of `go work sync`, I wonder if doing this is correct.

Reply:
I guess not?  I did notice many changes when I did go work sync.